packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,345 @@
+## Tagging with the CDK
+
+Tags are implemented using `Aspects` for details on the implementation see the
+detailed sections below.
+
+### Tag Quickstart
+
+Tags can be applied to any `Construct` in the CDK. By default the Tag will
+propagate to any resource that is a child of the construct and supports tagging.
+
+For example, if you create a stack and want anything in the stack to receive a
+tag:
+
+```ts
+import cdk = require('@aws-cdk/cdk');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest'));
+
+// any resources added that support tags will get them
+```
+
+### Advanced Tagging
+
+#### Resource Type Controls
+
+If you need to apply a tag, but want to specify which CloudFormation Resource
+Types are affected you can include or exclude those in properties.
+
+```ts
+import cdk = require('@aws-cdk/cdk');
```

Comment:
I don't think you need the whole program every time. Only the tagging related snippets will do:

```ts
this.apply(new cdk.Tag(..., {
  includeResourceTypes: [ ec2.CfnVpc.resourceType ]
});
```